x/observer/keeper/keeper_utils.go

```diff
@@ -8,6 +8,66 @@ import (
 	"github.com/zeta-chain/zetacore/x/observer/types"
 )
 
+func (k Keeper) AddVoteToBallot(ctx sdk.Context, ballot types.Ballot, address string, observationType types.VoteType) (types.Ballot, error) {
+	ballot, err := ballot.AddVote(address, observationType)
+	if err != nil {
+		return ballot, err
+	}
+	ctx.Logger().Info(fmt.Sprintf("Vote Added | Voter :%s, ballot idetifier %s", address, ballot.BallotIdentifier))
+	k.SetBallot(ctx, &ballot)
+	return ballot, err
+}
+
+// CheckIfFinalizingVote checks if the ballot is finalized in this block and if it is, it sets the ballot in the store
+// This function with only return true if the ballot moves for pending to success or failed status with this vote.
+// If the ballot is already finalized in the previous vote , it will return false
+func (k Keeper) CheckIfFinalizingVote(ctx sdk.Context, ballot types.Ballot) (types.Ballot, bool) {
+	ballot, isFinalized := ballot.IsBallotFinalized()
+	if !isFinalized {
+		return ballot, false
+	}
+	k.SetBallot(ctx, &ballot)
+	return ballot, true
+}
+
+// IsAuthorized checks whether a signer is authorized to sign , by checking their address against the observer mapper which contains the observer list for the chain and type
+func (k Keeper) IsAuthorized(ctx sdk.Context, address string, chain *common.Chain) (bool, error) {
+	observerMapper, found := k.GetObserverMapper(ctx, chain)
+	if !found {
+		return false, errors.Wrap(types.ErrNotAuthorized, fmt.Sprintf("observer list not present for chain %s", chain.String()))
+	}
+	for _, obs := range observerMapper.ObserverList {
+		if obs == address {
+			return true, nil
+		}
+	}
+	return false, errors.Wrap(types.ErrNotAuthorized, fmt.Sprintf("address: %s", address))
+}
+
+func (k Keeper) FindBallot(ctx sdk.Context, index string, chain *common.Chain, observationType types.ObservationType) (ballot types.Ballot, isNew bool, err error) {
```

Comment:
```suggestion
func (k Keeper) FindBallot(ctx sdk.Context, index string, chain *common.Chain, observationType types.ObservationType,Threshold sdk.Dec) (ballot types.Ballot, isNew bool, err error) {
```
The threshold defined in params is only for inbound and outbound TX , we can reuse it for Blame as well, but it would be better to maintain a list of thresholds in params .

Tss voter for example uses 100%  , https://github.com/zeta-chain/zeta-node/blob/2437225c0294635a5bd150e50d61853157b2e47d/x/crosschain/keeper/keeper_tss_voter.go#L61-L61

Reply:
We can make that change for sure but this is the first iteration of the blame implementation and I don't want to introduce this right now, it might take more time to test.